### ISSUE TITLE
Git for Windows OpenSSH8.2p1 update affects Azure DevOps Server 2019

### ISSUE THREAD

**Issue by vScherb** [MicrosoftDocs/azure-devops-docs#7519]
**Tested systems**: Azure DevOps Server 2019 1.1 (Dev17.M153.5), Azure Repos
**Affected system**: Azure DevOps Server 2019 1.1 (Dev17.M153.5)
**Git for Windows**: [2.25.1.windows.1](https://github.com/git-for-windows/git/releases/tag/v2.25.1.windows.1)

The patch note of the Git for Windows update states:
> Comes with [OpenSSH v8.2p1](https://www.openssh.com/txt/release-8.2).

Which removes "diffie-hellman-group14-sha1" from the default key exchange proposal.
> Potentially-incompatible changes
> ================================
> [...]
> \ * ssh(1), sshd(8): this release removes diffie-hellman-group14-sha1
> from the default key exchange proposal for both the client and
> server.
> [...]

With and updated system all server opertations on the on premises instance of Azure DevOps Server Update 1.1 are rejected with the message.
> $ git fetch
> Unable to negotiate with 10.\*.\*.\* port 22: no matching key exchange method found. Their
> offer: diffie-hellman-group1-sha1,diffie-hellman-group14-sha1
> fatal: Could not read from remote repository.
>
> Please make sure you have the correct access rights
> and the repository exists.

My assumption is that older versions are affected as well.
But I cannot verify that.

The documentation should inform users about this unexpected behavior for the affected Versions of Azure DevOps / TFS until an update for these products is released.

I tested this behavior with the Azure Repos (Azure DevOps online) this seems not to be affected by this issue.

In the developercommunity is already a [discussion](https://developercommunity.visualstudio.com/content/problem/921226/git-ssh-access-offers-weak-algorithms.html) about this topic.
The suggested [solution](https://developercommunity.visualstudio.com/content/problem/921226/git-ssh-access-offers-weak-algorithms.html#reply-921447), which enables this lecagy key exchange method, by [Aaron MASOVER](https://developercommunity.visualstudio.com/users/233239/0c63b858-2450-6a05-b4f8-990df1d43e9c.html) works.
``` bash
Host tfs.onpremiseshost.lan
	KexAlgorithms +diffie-hellman-group14-sha1
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 64e85ff9-9add-d15c-e92f-02d7dd05e0f2
* Version Independent ID: 0cbd7427-6acd-3cfc-1a3b-8bf5ca60f3bd
* Content: [Connect to your Git repos with SSH - Azure Repos](https://docs.microsoft.com/en-us/azure/devops/repos/git/use-ssh-keys-to-authenticate?view=azure-devops&tabs=current-page)
* Content Source: [docs/repos/git/use-ssh-keys-to-authenticate.md](https://github.com/MicrosoftDocs/vsts-docs/blob/master/docs/repos/git/use-ssh-keys-to-authenticate.md)
* Product: **devops**
* Technology: **devops-code-git**
* GitHub Login: @apawast
* Microsoft Alias: **sdanie**

**Comment by vScherb** [comment]
A new post was added by Microsoft
> We have released a fix for this issue! Thank you for providing valuable feedback to help improve the product.
>
> The fix I mentioned earlier is now available for everyone. As mentioned, the new algorithm is coming to the upcoming version of Azure DevOps Server. Looking at the impact this will have, we have also decided to make it available to 'n' and 'n-1' major version of the Server. However for these older versions, it is likely to take another month or so before its available as a patch.

As marked as the [Solution by Microsoft](https://developercommunity.visualstudio.com/content/problem/921226/git-ssh-access-offers-weak-algorithms.html#reply-container-941869).

As a helpful reference for the documentation visitors I keep this issue open until it is at least fixed for Azure DevOps 2019 (on premises).

**Comment by vScherb** [comment]
# Update 2019 Update 1.1 Patch 1

Yesterday we installed the [security patch (Azure DevOps Server 2019 Update 1.1 Patch 1 Release Date: March 10, 2020)](https://docs.microsoft.com/en-us/azure/devops/server/release-notes/azuredevops2019u1?view=azure-devops#azure-devops-server-2019-update-11-patch-1-release-date-march-10-2020) on our Azure DevOps server.

After the installation the issue still persists.
The workaround with the `~/.ssh/config` fixes the problem.

**Comment by valeriob** [comment]
Hi,
we have the same problem with Microsoft Visual Studio Team Foundation Server Version 15.117.27414.0 (Just patched with Update 3.1 Patch 12).
How do we enable sha2 Algorithms ?
